lti_consumer/lti_1p3/extensions/rest_framework/serializers.py

```diff
@@ -89,3 +89,137 @@ class Meta:
             'startDateTime',
             'endDateTime',
         )
+
+
+class LtiAgsScoreSerializer(serializers.ModelSerializer):
+    """
+    LTI AGS LineItemScore Serializer.
+
+    This maps out the internally stored LtiAgsScore to
+    the LTI-AGS API Specification, as shown in the example
+    response below:
+
+    {
+      "timestamp": "2017-04-16T18:54:36.736+00:00",
+      "scoreGiven" : 83,
+      "scoreMaximum" : 100,
+      "comment" : "This is exceptional work.",
+      "activityProgress" : "Completed",
+      "gradingProgress": "FullyGraded",
+      "userId" : "5323497"
+    }
+
+    Reference:
+    https://www.imsglobal.org/spec/lti-ags/v2p0#example-application-vnd-ims-lis-v1-score-json-representation
+    """
+
+    timestamp = serializers.DateTimeField()
+    # All 'scoreGiven' values MUST be positive numeric (including 0).
+    # 'scoreMaximum' represents the denominator and MUST be present when 'scoreGiven' is present
+    scoreGiven = serializers.FloatField(source='score_given', required=False, allow_null=True, default=None)
+    scoreMaximum = serializers.FloatField(source='score_maximum', required=False, allow_null=True, default=None)
+    comment = serializers.CharField(required=False, allow_null=True)
+    activityProgress = serializers.CharField(source='activity_progress')
+    gradingProgress = serializers.CharField(source='grading_progress')
+    userId = serializers.CharField(source='user_id')
+
+    def validate_scoreGiven(self, value):
+        if not self.instance and value is None:
+            raise serializers.ValidationError('Cannot delete score. No score currently exists for this user')
+        return value
+
+    def validate_scoreMaximum(self, value):
+        if self.initial_data.get('scoreGiven', None) and value is None:
+            raise serializers.ValidationError('Cannot have empty scoreMaximum when scoreGiven is set')
+        return value
+
+    def validate_timestamp(self, value):
+        if self.instance:
+            if self.instance.timestamp > value:
+                raise serializers.ValidationError('Score timestamp can only be updated to a later point in time')
+            elif self.instance.timestamp == value:
+                raise serializers.ValidationError('Score already exists for the provided timestamp')
+        return value
+
+    def update(self, instance, validated_data):
+        # When 'scoreGiven' is not present, this indicates there is presently no score for that user,
+        # and the platform should clear any previous score value it may have previously received from
+        # the tool and stored for that user and line item
+        if instance and validated_data.get('score_given', None) is None:
+            instance.delete()
+            return instance
+
+        return super().update(instance, validated_data)
```

Comment:
No, this is not the desired behavior.
Again, with the same request as before. If a user already has a grade, and this is received:
```
{
  "timestamp": "2017-04-16T18:54:36.736+00:00",
  "activityProgress" : "Completed",
  "gradingProgress" : "PendingManual",
  "userId" : "5323497"
}
```
You should just store `None` in score, but leave the other attributes.